### DISCUSSION TITLE
Error al mandar el xml al timbrar con Comercio Digital

### DISCUSSION THREAD

**Discussion by GwEeN-BeTtH** [bambucode/tfacturaelectronica#185] [Q&A] [answered]
Buen día, ps ya tengo varios dias batallando con el error de la digestión, en otra conversación vi que varios colegas batallaron con eso e intente replicar lo que ellos hicieron para poder resolver el error, pero no tuve éxito, cheque el xslt y xsd todo bien, mas siempre me dice lo mismo, creí que podía ser la codificación utf-8, pero checando mi xml en validadores de cfdi 4.0 online me indican que la codificación es la correcta además de que en el notepad++ me indica tmb que esta correcto, ya no encuentro que mas hacer, alguna idea de que pueda ser??.. Uso solo el complemento de la carta porte 3.0 pero aunque no se lo incluya el error persiste.. Cabe mencionar que los validadores me sacan la cadena original la cual comparo con la que saca la librería y esta identica, ningún espacio de mas ni nada raro.. 

"El resultado de la digestión debe ser igual a la desencripción del sello" =(

De antemano, muchas gracias =)

**Comment by ptorres** [comment]
hola, no tenemos incorporado todavía el complemento de carta porte a la librería, imagino que tu lo desarrollaste ya, si gustas adjuntar tu ejemplo para ver si puedo apoyarte con tu error.

**Reply by GwEeN-BeTtH** [reply]
[Facturacion.ComplementoCartaPorte30.txt](https://github.com/bambucode/tfacturaelectronica/files/14133472/Facturacion.ComplementoCartaPorte30.txt)

Hola, claro, te adjunto el archivo, en realidad todo lo hizo el databinding de delphi, yo solo le hice unas adecuaciones para que cuando el atributo no tuviera un valor, no lo incluyera en el xml y otros detallitos.. No se que mas ocupes como ejemplo para checarlo, igual me avisas, pero como ya comentaba anteriormente, aunque no le adjunte el complemento de la carta porte, y el xml tipo traslado vaya sin este, no me lo timbra así que el error debe ser general, adjunto tmb el xml que genero, sin el complemento.. lo verifique en validadores online y me dicen que mi codificación está bien, la estructura, pero tmb me marca el mismo error =/ ... Sospecho que pudiera ser del lado de la digestión, cuando encripto la cadena original, pero pues todo eso lo hace ya la librería y no tendría porque estar mal.. en fin...  

Agradezco la atención! =)

[compDesdeAdmon.xml.txt](https://github.com/bambucode/tfacturaelectronica/files/14133503/compDesdeAdmon.xml.txt)

**Comment by GwEeN-BeTtH** [comment, marked as answer]
Lo he resuelto =) ...
Resulta que al crear la cadena, no se porque, la función de la librería le agregaba espacios y saltos de línea en hexadecimal, lo cual entorpecía la encripción y desencripción del sello al mandar timbrar. Limpiando la cadena original se corrige el error y timbra perfecto, con o sin carta porte =D... Igual surgen detalles cuando típicamente los nombres de los artículos tienen comillas simples o caracteres raros, pero esto pasa por la codificación del XML.. ya que no se porque las funciones que se supone aseguran esta codificación (Utf-8) al hacer conversiones en repetidas ocasiones, lo cual no se logra, porque?? ni idea, lo que hago es una pequeña mexicanada.. guardo el xml en un archivo de texto, con esta codificación, al guardarlo y volverlo a cargar, ya toma en cuenta la codificación y todo perfecto, no mas errores de digestión.. 

Muchas gracias por la atención a todos! cualquier cosa estoy a la orden!
